the-collab-lab/tcl-69-smart-shopping-list
# [MF] ShareListDialog to share list on List page

## Description

This code adds a `ShareListDialog` component that is rendered on the List page. 
It has a nested `ShareEmailInput` which is passed the `setRecipientEmail` value using the email inputted by user. 
After entering an email in the Dialog, the user will be alerted on the List page of success or failure in the same manner as the /manage-list endpoint. 
The "shareList" function is the same one imported from `firebase.js` file

It also addresses case sensitivity in the email entered into the input adding the .toLowerCase() method.

There are a couple of updates things that I can suggest being refactored: 

- handling of the status responses from `shareList` function directly in the Dialog.
- deleting of previous value in the email input when returning to Dialog. 
- the css styling both the actual styling but also we should determine our approach for styling and if we plan to use any framework or library.

## Related Issue

closes #34 

## Acceptance Criteria

- [x] Integrate @mxmason 's `Dialog` component into project.
- [x] Add `Dialog` component to List page to share list and/or add new item. 
- [x] Pass the `shareList` function to the `Dialog` component as an option with input for email.
- [x] Ensure there is a `cancel` button to leave `Dialog` without submitting request

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

## Updates

### Before

![Screenshot 2024-03-19 at 5 45 21 PM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/65b02869-4dca-4098-839a-434324bcdb89)

### After

#### With Share List Button
![Screenshot 2024-03-19 at 5 45 52 PM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/ecddd5b5-34b1-4a2e-be77-563d0bd9b510)

#### Share List Dialog is open

![Screenshot 2024-03-19 at 5 46 40 PM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/26ee046b-4f1d-40ca-b907-799c6fbbf88a)

#### After invalid user email is entered

![Screenshot 2024-03-19 at 5 47 27 PM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/c9a716a2-6407-481c-be32-e92c1603782c)

#### After valid user email entered but previously shared

![Screenshot 2024-03-19 at 5 48 40 PM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/0dae67f5-bd85-44bc-b91c-1a4b6c95c627)

#### After valid user email entered and success

![Screenshot 2024-03-19 at 5 50 01 PM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/4623c7b3-7d40-46e3-8ce8-59619bb6f862)


## Testing Steps / QA Criteria

After login navigate to list you have not shared.
Enter email of valid user and select 'confirm'.
Trying entering the same email.
Trying entering a non valid user email.
